images/build/php/build.sh

```diff
@@ -0,0 +1,161 @@
+#!/bin/bash
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT license.
+# --------------------------------------------------------------------------------------------
+
+set -eux
+
+START_TIME=`date +%s`
+
+PHP_MAJOR=${PHP_VERSION:0:1}
+PHP_MINOR=${PHP_VERSION:2:1}
+INSTALLATION_PREFIX="/opt/php/$PHP_VERSION"
+PHP_INI_DIR="$INSTALLATION_PREFIX/ini"
+PHP_SRC_DIR="/usr/src/php"
+
+mkdir -p "$PHP_INI_DIR/conf.d";
+
+# Apply stack smash protection to functions using local buffers and alloca()
+# Make PHP's main executable position-independent (improves ASLR security mechanism, and has no performance impact on x86_64)
+# Enable optimization (-O2)
+# Enable linker optimization (this sorts the hash buckets to improve cache locality, and is non-default)
+# Adds GNU HASH segments to generated executables (this is used if present, and is much faster than sysv hash; in this configuration, sysv hash is also generated)
+# https://github.com/docker-library/php/issues/272
+PHP_CFLAGS="-fstack-protector-strong -fpic -fpie -O2"
+PHP_CPPFLAGS="$PHP_CFLAGS"
+PHP_LDFLAGS="-Wl,-O1 -Wl,--hash-style=both -pie"
+
+PHP_URL="https://secure.php.net/get/php-$PHP_VERSION.tar.xz/from/this/mirror"
+PHP_ASC_URL="" # "https://secure.php.net/get/php-$PHP_VERSION.tar.xz.asc/from/this/mirror"
+GPG_KEYS=($GPG_KEYS) # Cast the string to an array
+PHP_MD5=""
+
+
+fetchDeps='wget';
+if ! command -v gpg > /dev/null; then
+	fetchDeps="$fetchDeps dirmngr gnupg"
+fi
+
+apt-get update && apt-get install -y --no-install-recommends $fetchDeps
+rm -rf /var/lib/apt/lists/*
+
+mkdir -p /usr/src
+cd /usr/src
+wget -O php.tar.xz "$PHP_URL"
+
+if [ -n "$PHP_SHA256" ]; then
+	echo "$PHP_SHA256 *php.tar.xz" | sha256sum -c -;
+fi;
+if [ -n "$PHP_MD5" ]; then
+	echo "$PHP_MD5 *php.tar.xz" | md5sum -c -;
+fi;
+
+if [ -n "$PHP_ASC_URL" ]; then
+	wget -O php.tar.xz.asc "$PHP_ASC_URL";
+	export GNUPGHOME="$(mktemp -d)";
+	for i in {1..5}; do
+		gpg --batch --keyserver ha.pool.sks-keyservers.net --recv-keys "${GPG_KEYS[0]}" || \
+		gpg --batch --keyserver ha.pool.sks-keyservers.net --recv-keys "${GPG_KEYS[1]}";
+		if [ $? -eq 0 ]; then break; fi
+    done
+	gpg --batch --verify php.tar.xz.asc php.tar.xz;
+	command -v gpgconf > /dev/null && gpgconf --kill all;
+	rm -rf "$GNUPGHOME";
+fi;
+
+apt-get purge -y --auto-remove -o APT::AutoRemove::RecommendsImportant=false $fetchDeps
+
+versionDevReqs='libssl-dev'
+if [ $PHP_MAJOR == '5' ]; then
+	versionDevReqs='libssl1.0-dev'
+fi
+
+savedAptMark="$(apt-mark showmanual)";
+apt-get update;
+apt-get install -y --no-install-recommends \
```

Comment:
These are being installed for each of the php versions. Can they be moved to the pre-req image step `php-build-prereqs` and this way its executed only once